src/widgets/models/CategoryModel.js

```diff
@@ -1,33 +1,48 @@
 import { executeSQL } from '../../api';
-import { filtersToSQL, viewportToSQL } from '../../api/FilterQueryBuilder';
+import { filtersToSQL } from '../../api/FilterQueryBuilder';
+import {groupValuesByColumn} from '../operations/grouping';
 
 export const getCategories = async (props) => {
-  const { data, credentials, column, operation, filters, viewport, opts } = props;
+  const { data, credentials, column, operation, filters, viewport, dataSource, viewportFilter, viewportFeatures, type, opts } = props;
 
   const operationColumn = props.operationColumn || column;
 
   if (Array.isArray(data)) {
     throw new Error('Array is not a valid type to get categories');
   }
 
-  let query =
-    (viewport && `SELECT * FROM (${data})  as q WHERE ${viewportToSQL(viewport)}`) ||
-    data;
+  if (type === 'bq' && !viewportFilter) {
+    throw new Error('Category Widget error: BigQuery layers needs "viewportFilter" prop set to true.');
+  }
+
+  // It's an await because we probably will move this calculation need to a webworker
+  if (viewportFilter) {
+    const targetFeatures = viewportFeatures[dataSource];
```

Comment:
This method shouldn't have logic around the source. dataSource variable is not required